pkg/webhook/server/webhook_cert_init.go

```diff
@@ -0,0 +1,228 @@
+	} else {
+		log.Info("Created secure service")
+	}
+	return nil
+}
+
+func getSecureServiceSpec(namespace string) *v1.Service {
+	label := map[string]string{"app": "che-workspace-controller"}
+
+	port := int32(443)
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "workspace-controller",
+			Namespace: namespace,
+			Labels:    label,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/serving-cert-secret-name": "workspace-controller",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Port:       port,
+					Protocol:   "TCP",
+					TargetPort: intstr.FromString("webhook-server"),
+				},
+			},
+			Selector: label,
+		},
+	}
+
+	return service
+}
+
+func syncConfigMap(ctx context.Context, crclient client.Client, namespace string) error {
+	secureConfigMap := getSecureConfigMapSpec(namespace)
+	if err := crclient.Create(ctx, secureConfigMap); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+		existingCfg := &v1.Service{}
+		err := crclient.Get(ctx, types.NamespacedName{
+			Name:      secureConfigMap.Name,
+			Namespace: secureConfigMap.Namespace,
+		}, existingCfg)
+
+		err = crclient.Update(ctx, secureConfigMap)
+		if err != nil {
+			return err
+		}
+		log.Info("Updated secure configmap")
+	} else {
+		log.Info("Created secure configmap")
+	}
+	return nil
+}
+
+func getSecureConfigMapSpec(namespace string) *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secureServiceName,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+	}
+}
+
+// Update the deployment with the volumes needed for webhook server if they aren't already present
+func updateDeployment(ctx context.Context, crclient client.Client, namespace string) error {
+	deployment, err := controller.FindControllerDeployment(ctx, crclient)
+	if err != nil {
+		return err
+	}
+
+	isVolumeMissing := appendVolumeIfMissing(&deployment.Spec.Template.Spec.Volumes,
+		v1.Volume{
+			Name: "webhook-tls-certs",
+			VolumeSource: v1.VolumeSource{
+				Projected: &v1.ProjectedVolumeSource{
+					Sources: []v1.VolumeProjection{
+						{
+							ConfigMap: &v1.ConfigMapProjection{
+								LocalObjectReference: v1.LocalObjectReference{
+									Name: secureServiceName,
+								},
+								Items: []v1.KeyToPath{
+									{
+										Key:  "service-ca.crt",
+										Path: "./ca.crt",
+									},
+								},
+							},
+						},
+						{
+							Secret: &v1.SecretProjection{
+								LocalObjectReference: v1.LocalObjectReference{
+									Name: "workspace-controller",
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+
+	isVMMissing := appendVolumeMountIfMissing(&deployment.Spec.Template.Spec.Containers[0].VolumeMounts,
+		*&v1.VolumeMount{
+			Name:      "webhook-tls-certs",
```

Comment:
Such duplicated hard-coded values are error-prone. I propose to move configuration objects to separate file and move everything (or at least that is used more than once) to const
See https://github.com/sleshchenko/che-workspace-operator/commit/5ca6b6d56c14c2548fa03cee61cef73ae6a68c8c